tensorflow_addons/custom_ops/layers/cc/kernels/correlation_cost_op_gpu.cu.cc

```diff
@@ -69,19 +71,20 @@ __global__ void pad_and_no_transpose(const float *input, float *output, int C,
   const int pW = (W + 2 * P);
   const int pH = (H + 2 * P);
 
-  float value;
   for (int c = c0; c < C; c += THREADS_PER_BLOCK) {
-    value = input[n * (C * H * W) + h * (W * C) + w * C + c];
-    output[n * (C * pH * pW) + (h + P) * (pW * C) + (w + P) * C + c] = value;
+    output[n * (C * pH * pW) + (h + P) * (pW * C) + (w + P) * C + c] =
+        ldg(input + n * (C * H * W) + h * (W * C) + w * C + c);
   }
 }
 
 template <unsigned int THREADS_PER_BLOCK>
-__global__ void Correlation_forward(float *output, int Cout, int Hout, int Wout,
-                                    float *pInput1, int Cin, int Hin, int Win,
-                                    float *pInput2, int pad, int kernel_size,
-                                    int max_displacement, int stride1,
-                                    int stride2) {
+__global__ void Correlation_forward(float *__restrict__ output, int Cout,
```

Comment:
We are not supporting Visual c++ right?

Reply:
@bhack Sorry for the late reply. We currently only build GPU ops linux wheel. https://github.com/tensorflow/addons/issues/784

Reply:
OK cause in TF lite we had guards in multiple places lke https://github.com/tensorflow/tensorflow/blob/master/tensorflow/lite/kernels/internal/optimized/sse_tensor_utils_impl.h#L22